src/components/pages/WelcomePage.tsx

```diff
@@ -183,21 +185,28 @@ class WelcomePage extends React.Component<Props, State> {
         linkedEventsPost: linkedEventsPost
       });
     });
-  
-    await this.getPopularPagesImageUrl();
+
+    this.getPopularPagesImageUrl();
   }
 
   /**
    * Fetch PopularPages featured image URL 
    */
   private getPopularPagesImageUrl = () => {
     const { popularPages } = this.state;
-    return popularPages.map(async(page) => {
+    const api = ApiUtils.getApi();
+    const result : PageWithImgUrl[] = [];
+
+    popularPages.map(async (page) => {
```

Comment:
```suggestion
    popularPages.forEach(async (page) => {
```

Reply:
array.map() returns an array of items that were returned in the mapping function. When using map you could return the entire map in result variable and instead of pushing to result you would return the item in map function. When using forEach() nothing is returned but similar function that handles all array items one at a time is executed.